src/routes/docs/products/functions/develop-locally/+page.markdoc

```diff
@@ -71,6 +75,81 @@ You can use headers like dynamic API keys in your function, which give you acces
 Learn more about dynamic API keys
 {% /arrow_link %}
 
+# Hot reloading {% #hot-reloading %}
+
+You can hot reload your locally developed functions, which means you can update the function code and see the changes happen in real time without restarting. Compiled languages must translate the source code into machine code, causing the system to rebuild continuously. However, interpreted languages only rebuild when you change the dependency file. Otherwise, updates can be made through a file swap.
```

Comment:
`Otherwise, updates can be made through a file swap.`

What does this mean? I'm not sure I understand it

Reply:
The file swap happens inside the `.appwrite` folder.
- Appwrite puts files from last build (extract it from `tar.gz`)
- Appwrite put files from your function folder there (allows override)
- Appwrite make a new build `tar.gz`
- Appwrite use that to start container again

It's "faking" build step without actually building - just updating files in build file
Regarding Docker container and it's files, Matej believes each container has a folder in /var directory. At least on Linux, no idea on Windows. but thats too low-level. Matej doesn't think we need to explain to this extend though.